File: docs/competitors/C++/GCD.cpp
#include <iostream>
#include <limits>
#include <string>

int gcd(int a, int b)
{
    while (b != 0)
    {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int readInt(const std::string &prompt)
{
    int value;
    while (true)
    {
        std::cout << prompt;
        if (std::cin >> value)
        {
            return value;
        }
        else
        {
            std::cout << "Error: please enter an integer." << std::endl;
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
        }
    }
}

int main()
{
    int x = readInt("Enter the first integer: ");
    int y = readInt("Enter the second integer: ");

    std::cout << "GCD = " << gcd(x, y) << std::endl;
    return 0;
}

File: docs/competitors/Python/GCD.py
def gcd(a: int, b: int):

    while b != 0:
        a, b = b, a % b
    return a

def read_int(prompt: str):
    while True:
        try:
            value = int(input(prompt))
            return value
        except ValueError:
            print("Error: please enter an integer.")

def main():
    x = read_int("Enter the first integer: ")
    y = read_int("Enter the second integer: ")

    print(f"GCD = {gcd(x, y)}")

if __name__ == "__main__":
    main()
